[ddsql] Add examples of multi-valued tags handling to DDSQL tags documentation (#31292)
DataDog/documentation: content/en/ddsql_reference/_index.md

GROUP BY instance_type
```

> Tags are key-value pairs where each key can have zero, one, or multiple tag values corresponding to it. When accessed, the tag value returns a single string, containing _all_ corresponding values. When the data had multiple tag values for the same tag key, these will be represented as a sorted, comma-separated string. For example:

Reviewer: ```suggestion
Tags are key-value pairs where each key can have zero, one, or multiple tag values corresponding to it. When accessed, the tag value returns a single string, containing _all_ corresponding values. When the data has multiple tag values for the same tag key, they are represented as a sorted, comma-separated string. For example:
```
